### PR TITLE
Handle content-type header with charset correctly

### PR DESCRIPTION
**Pull request by kibertoad**
We've hit another edge case where service was setting content-type with charset, and this was breaking validation.

### CODE REVIEW

**Comment by kibertoad** [comment]
@kobik What do you think?

**Comment by kobik** [comment]
@kibertoad, I'm OK with that.

Can you sync with master first?

**Comment by kibertoad** [comment]
@kobik Done!

**Comment by kobik** [comment]
@kibertoad , i've the test so it won't rely on the encoding as part as the content-type.

you can sync again with master.

**Comment by kibertoad** [comment]
@kobik Thanks, updated!

File: src/validators/ResponseValidator.js [hunk 1]
```diff
@@ -17,7 +17,8 @@ function responseValidator(response, data) {
         if (bodySchema.validate) {
             validator = bodySchema;
         } else {
-            const responseContentType = data.headers['Content-Type'] || data.headers['content-type'] || DEFAULT_CONTENT_TYPE;
+            const responseContentTypeHeader = data.headers['Content-Type'] || data.headers['content-type'] || DEFAULT_CONTENT_TYPE;
+            const responseContentType = responseContentTypeHeader.split(';')[0]; // This is to filter out things like charset
```
Comment (kobik) [inline src/validators/ResponseValidator.js]: maybe it's better to also trim the result of the result of the split?

Comment (kibertoad) [reply src/validators/ResponseValidator.js]: done

**Pull request merged**